packages/block-library/src/bookmark/index.php

```diff
@@ -0,0 +1,170 @@
+				'userLikedPosts' => $liked_posts,
+			)
+		);
+
+		wp_enqueue_script( 'core-bookmark-user-data' );
+	}
+	return sprintf(
+		"<div
+		data-wp-interactive='core/bookmark'
+		%s
+		%s
+	>
+		<div
+			data-wp-on--click='actions.toggleBookmark'
+			data-wp-class--bookmark-active='state.isBookmarked'
+		>
+		<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' width='24' height='24' aria-hidden='true' focusable='false' fill='none' stroke='currentColor' stroke-width='2'><path d='%s'></path></svg>
+		</div>
+	</div>
+	",
+		$wrapper_attributes,
+		$context_data,
+		$icon
+	);
+}
+
+/**
+ * Registers the `bookmark` block, new user meta & rewrite rules.
+ *
+ * @since 6.8.0
+ */
+function register_block_core_bookmark() {
+	register_meta(
+		'user',
+		'liked_posts',
+		array(
+			'type'              => 'array',
+			'single'            => true,
+			'show_in_rest'      => array(
+				'schema' => array(
+					'type'  => 'array',
+					'items' => array( 'type' => 'integer' ),
+				),
+			),
+			'sanitize_callback' => function ( $value ) {
+				return is_array( $value ) ? array_map( 'sanitize_text_field', $value ) : array();
+			},
+			'auth_callback'     => function ( $allowed, $meta_key, $user_id ) {
+				return get_current_user_id() === (int) $user_id;
+			},
+		)
+	);
+
+	register_block_type_from_metadata(
+		__DIR__ . '/bookmark',
+		array(
+			'render_callback' => 'render_block_core_bookmark',
+		)
+	);
+
+	add_rewrite_rule( '^favorites/?$', 'index.php?pagename=favorites', 'top' );
+	add_rewrite_rule( '^favorites/page/([0-9]+)/?$', 'index.php?pagename=favorites&paged=$matches[1]', 'top' );
+	flush_rewrite_rules();
+}
+
+add_action( 'init', 'register_block_core_bookmark' );
+
+/**
+ * Includes the all-bookmarked template.
+ *
+ * @since 6.8.0
+ */
+function block_core_bookmark_load_favorites_template( $template ) {
+	global $wp;
+	if ( isset( $wp->query_vars['pagename'] ) && 'favorites' === $wp->query_vars['pagename'] ) { // Ensure it's set
+		status_header( 200 );
+		if ( ! is_user_logged_in() ) {
+			wp_register_script( 'core-bookmarks-logged-out-user', '/wp-content/plugins/gutenberg/packages/block-library/src/bookmark/bookmarks-logged-out-user.js', array(), filemtime( '/wp-content/plugins/gutenberg/packages/block-library/src/bookmark/bookmarks-logged-out-user.js' ), true );
+			wp_enqueue_script( 'core-bookmarks-logged-out-user' );
+		}
+		return 'wp-content/plugins/gutenberg/packages/block-library/src/bookmark/all-bookmarked.php';
+	}
+	return $template;
+}
+add_filter( 'template_include', 'block_core_bookmark_load_favorites_template' );
+
+/**
+ * Sets the page name.
+ *
+ * @since 6.8.0
+ */
+function block_core_bookmark_set_favorites_page_title( $title ) {
+	global $wp;
+	if ( isset( $wp->query_vars['pagename'] ) && 'favorites' === $wp->query_vars['pagename'] ) {
+		$title['title'] = 'Favorite Posts';
+	}
+	return $title;
+}
+add_filter( 'document_title_parts', 'block_core_bookmark_set_favorites_page_title' );
```

Comment:
I'm guessing this updates the document title tag. All human-readable user-facing strings need to be translated.